src/gui/sdlmain.cpp

```diff
@@ -1924,23 +1936,61 @@ static bool DetectResizableWindow()
 #endif // C_OPENGL
 }
 
+static bool detect_shader_fixed_size()
+{
+#if C_OPENGL
+	if (sdl.desktop.want_type != SCREEN_OPENGL)
+		return false;
+
+	const std::string sname = get_glshader_value();
+	return (sname == "crt-fakelottes-flat" || sname == "crt-easymode-flat");
+#else
+	return false;
+#endif // C_OPENGL
+}
+
+static std::tuple<int, int> detect_window_size()
+{
+	SDL_Rect bounds;
+	SDL_GetDisplayBounds(sdl.display_number, &bounds);
+	constexpr int resolutions[][2] = {
+	        {1600, 1200},
+	        {1280, 960},
+	        {1024, 768},
+	        {800, 600},
+	};
+	for (const auto &wh : resolutions) {
+		if (bounds.w > wh[0] && bounds.h > wh[1])
+			return std::make_tuple(wh[0], wh[1]);
+	}
+	return std::make_tuple(640, 480);
```

Comment:
It's unfortunate SDL doesn't provide a simple `SDL_dims` object with width and height members. It only provide `SDL_rect`.
This code could just be cleaner `dim.w`, `dim.h`, etc.. and `return Dim(w, h)` instead of `make_tuple(wh[0], wh[1])`.. which feels cumbersome.

Reply:
With SDL_Point, being a simple POD struct `return {640, 480};` should work - let's see if it passes CI :)

Reply:
Working nicely :-)